[CI] StoredFieldsSequentialIT class failing

**Build Scans:**
- [elasticsearch-intake #22476 / part3](https://gradle-enterprise.elastic.co/s/7rhrxikiad2qs)
- [elasticsearch-pull-request #69962 / part-3](https://gradle-enterprise.elastic.co/s/dpql47fxvzvyg)
- [elasticsearch-pull-request #69936 / part-3](https://gradle-enterprise.elastic.co/s/ugwsauz2sisaa)
- [elasticsearch-intake #22452 / part3](https://gradle-enterprise.elastic.co/s/tpfl5hmajtkz2)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:single-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.single_node.StoredFieldsSequentialIT.testAggTwentyPercent" -Dtests.seed=F14D36542636DDA6 -Dtests.locale=ar-SA -Dtests.timezone=Pacific/Guadalcanal -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.single_node.StoredFieldsSequentialIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(org.elasticsearch.xpack.esql.qa.single_node.StoredFieldsSequentialIT),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: a map containing
errors: <false>
     but: a map containing
errors: expected <false> but was <true>
  took: <200> unexpected but ok
 items: <[{create={_index=test, _id=dWGWhpYBSo8q6TWHwLyI, error={reason=[1:23] failed to parse: [1:30] Invalid UTF-8 start byte 0xa0
 at [Source: (byte[])"{"test":"test٠٠٠", "i": ٠}"; line: 1, column: 30], caused_by={reason=[1:30] Invalid UTF-8 start byte 0xa0
 at [Source: (byte[])"{"test":"test٠٠٠", "i": ٠}"; line: 1, column: 30], caused_by={reason=Invalid UTF-8 start byte 0xa0
 at [Source: (byte[])"{"test":"test٠٠٠", "i": ٠}"; line: 1, column: 30], type=json_parse_exception}, type=x_content_parse_exception}, type=document_parsing_exception}, status=400}}, {create={_index=test, _id=dmGWhpYBSo8q6TWHwLyJ, error={reason=[1:23] failed to parse: [1:30] Invalid UTF-8 start byte 0xa1
 at [Source: (byte[])"{"test":"test٠٠١", "i": ١}"; line: 1, column: 30], caused_by={reason=[1:30] Invalid UTF-8 start byte 0xa1
 at [Source: (by
[truncated]
```

**Issue Reasons:**
- [main] 4 failures in class org.elasticsearch.xpack.esql.qa.single_node.StoredFieldsSequentialIT (6.5% fail rate in 62 executions)
- [main] 2 failures in step part3 (13.3% fail rate in 15 executions)
- [main] 2 failures in step part-3 (11.1% fail rate in 18 executions)
- [main] 2 failures in pipeline elasticsearch-intake (13.3% fail rate in 15 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (11.1% fail rate in 18 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/esql/qa/server/single-node/src/javaRestTest/java/org/elasticsearch/xpack/esql/qa/single_node/StoredFieldsSequentialIT.java
+++ b/x-pack/plugin/esql/qa/server/single-node/src/javaRestTest/java/org/elasticsearch/xpack/esql/qa/single_node/StoredFieldsSequentialIT.java
@@ -184,7 +184,7 @@ public class StoredFieldsSequentialIT extends ESRestTestCase {
         bulk.addParameter("refresh", "");
         StringBuilder b = new StringBuilder();
         for (int i = 0; i < 1000; i++) {
-            b.append(String.format("""
+            b.append(String.format(Locale.ROOT, """
                 {"create":{"_index":"test"}}
                 {"test":"test%03d", "i": %d}
                 """, i, i));

